Regenerated SDK with latest draft auto-rest Swagger and correct the unit test

lebronJ opened the pull request.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/BaseTests.cs:
```diff
@@ -24,18 +24,20 @@ static BaseTests()
-            IComputerVisionAPI client;
-            client = new ComputerVisionAPI(new ApiKeyServiceClientCredentials(ComputerVisionSubscriptionKey), handlers: handler);
-            client.AzureRegion = Vision.AzureRegions.Westcentralus;
+            IComputerVisionAPI client = new ComputerVisionAPI(new ApiKeyServiceClientCredentials(ComputerVisionSubscriptionKey), handlers: handler)
```
Better to split the test for ComputerVision and Face, I see no gain to merge them together.
Two teams working on the same file will bring high potential of conflicts.

Suggest to have same structure as the generated SDK, which means move it here:
```
src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/
```

lebronJ replied:
Done, please help review this

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/FaceVerificationTests.cs:
```diff
@@ -72,14 +77,13 @@ public void FaceVerificationNegative2()
                 try
                 {
-                    CreatePersonResult createPersonResult = client.Person.CreateAsync(personGroupId, "David").Result;
+                    Person createPersonResult = client.Person.CreateAsync(personGroupId, "David").Result;
```
s/createPersonResult/person/g

Comment style for your reference: https://www.gnu.org/software/sed/manual/html_node/The-_0022s_0022-Command.html

lebronJ replied:
I see, thx.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/FaceAPI.cs:
```diff
@@ -58,19 +58,19 @@ public partial class FaceAPI : ServiceClient<FaceAPI>, IFaceAPI
+        /// Gets the IPersonOperations.
         /// </summary>
-        public virtual IPerson Person { get; private set; }
+        public virtual IPersonOperations Person { get; private set; }
```
s/IPersonOperations Person/IPersonGroupPersonOperations PersonGroupPerson/g

lebronJ replied:
Done
For changing all Person to PersonGroupPerson in FaceAPI/IFaceAPI:
There is no trailing `Operations` because there is no `PersonGroupPerson` object type but `Person`.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/FaceListOperations.cs:
```diff
@@ -58,11 +58,11 @@ public FaceList(FaceAPI client)
         /// </param>
         /// <param name='name'>
-        /// Name of the face list, maximum length is 128.
+        /// Name of the face collection, maximum length is 128.
```
Revert all these back. Face Collection is the an ancient name for FaceList.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IFaceAPI.cs:
```diff
@@ -54,19 +54,19 @@ public partial interface IFaceAPI : System.IDisposable
+        /// Gets the IPersonOperations.
         /// </summary>
-        IPerson Person { get; }
+        IPersonOperations Person { get; }
```
s/IPersonOperations Person/IPersonGroupPersonOperations PersonGroupPerson/g

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IFaceOperations.cs:
```diff
@@ -144,10 +144,7 @@ public partial interface IFaceOperations
-        /// Thrown when a required parameter is null
-        /// </exception>
-        Task<HttpOperationResponse<VerifyResult>> VerifyWithHttpMessagesAsync(string faceId1, string faceId2, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<HttpOperationResponse<VerifyResult>> VerifyWithHttpMessagesAsync(System.Guid faceId1, System.Guid faceId2, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
s/VerifyWithHttpMessagesAsync/VerifyFaceToFaceWithHttpMessagesAsync/g

Just make it specific to avoid potential misleading.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IFaceOperations.cs:
```diff
@@ -184,7 +181,7 @@ public partial interface IFaceOperations
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<HttpOperationResponse<IList<DetectedFace>>> DetectWithHttpMessagesAsync(string url, bool? returnFaceId = true, bool? returnFaceLandmarks = false, IList<FaceAttributeTypes> returnFaceAttributes = default(IList<FaceAttributeTypes>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<HttpOperationResponse<IList<DetectedFace>>> DetectWithHttpMessagesAsync(string url, bool? returnFaceId = true, bool? returnFaceLandmarks = false, IList<FaceAttributeType> returnFaceAttributes = default(IList<FaceAttributeType>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
DetectWithUrl?

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IFaceOperations.cs:
```diff
@@ -253,6 +250,6 @@ public partial interface IFaceOperations
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<HttpOperationResponse<IList<DetectedFace>>> DetectInStreamWithHttpMessagesAsync(Stream image, bool? returnFaceId = true, bool? returnFaceLandmarks = false, IList<FaceAttributeTypes> returnFaceAttributes = default(IList<FaceAttributeTypes>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<HttpOperationResponse<IList<DetectedFace>>> DetectInStreamWithHttpMessagesAsync(Stream image, bool? returnFaceId = true, bool? returnFaceLandmarks = false, IList<FaceAttributeType> returnFaceAttributes = default(IList<FaceAttributeType>), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
It should be OK to name it `DetectWithStream` so as to keep consistency.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IPersonGroupOperations.cs:
```diff
@@ -18,23 +18,23 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face
 
     /// <summary>
-    /// PersonGroup operations.
+    /// PersonGroupOperations operations.
```
Is this comments changeable? Just too wired to have duplicate operations.
If it is not easy achievable, please ignore this issue.

lebronJ replied:
It's because we have object called PersonGroup, no easy naming solution to change this.

huxuan replied:
resolved.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IPersonGroupOperations.cs:
```diff
@@ -18,23 +18,23 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face
         /// </param>
         /// <param name='name'>
-        /// Name of the face list, maximum length is 128.
+        /// Name of the face collection, maximum length is 128.
```
Seems we need a general description for the Name. This is not for face collection.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IPersonGroupOperations.cs:
```diff
@@ -91,21 +91,21 @@ public partial interface IPersonGroup
         /// <param name='userData'>
-        /// Optional user defined data for the face list. Length should not
-        /// exceed 16KB.
+        /// Optional user defined data for the face collection. Length should
```
Ditto. Same error. Not face collection here. Please also check other places. No similar comments again.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/IPersonOperations.cs:
```diff
@@ -19,22 +19,22 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face
+    /// PersonOperations operations.
     /// </summary>
-    public partial interface IPerson
+    public partial interface IPersonOperations
```
IPersonGroupPersonOperations

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/SimilarFace.cs:
```diff
@@ -55,36 +57,32 @@ public SimilarFaceResult(string faceId = default(string), string persistedFaceId
         /// </summary>
         [JsonProperty(PropertyName = "faceId")]
-        public string FaceId { get; set; }
+        public System.Guid? FaceId { get; set; }
```
FaceId? I do not think we have it here.

lebronJ replied:
FaceId is [returned](https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395237) when requested by faceIds.

huxuan replied:
Resolved.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/SimilarFace.cs:
```diff
@@ -55,36 +57,32 @@ public SimilarFaceResult(string faceId = default(string), string persistedFaceId
         /// </summary>
         [JsonProperty(PropertyName = "persistedFaceId")]
-        public string PersistedFaceId { get; set; }
+        public System.Guid? PersistedFaceId { get; set; }
```
nullable?

lebronJ replied:
Same as the above comment. 
And if we don't use method overload, PersistedFaceId is nullable because faceId will return when the FindSimilarRequest's target are faceIds.
Checked and fixed all guid's nullable properties in new pushed changes.

huxuan replied:
OK, this convince me well. Let's keep it currently.

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/TrainingStatus.cs:
```diff
@@ -11,62 +11,93 @@
+        /// person group created time.
+        /// </summary>
+        [JsonProperty(PropertyName = "createdDateTime")]
+        public System.DateTime? Created { get; set; }
```
nullable?

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/TrainingStatus.cs:
```diff
@@ -11,62 +11,93 @@
+        /// null value when the person group is not successfully trained.
+        /// </summary>
+        [JsonProperty(PropertyName = "lastActionDateTime")]
+        public System.DateTime? LastAction { get; set; }
```
nullable?

huxuan replied:
Seems it is possible to have nullable `LastAction`, but I do not think we need nullable `Created`.

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/Models/UpdatePersonFaceRequest.cs:
```diff
@@ -17,24 +17,22 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face.Models
-        /// size limit is 1KB</param>
-        public UpdatePersonFaceDataRequest(string userData = default(string))
+        /// size limit is 1KB.</param>
+        public UpdatePersonFaceRequest(string userData = default(string))
```
UserDataContract. As we will use it later in LargeFaceList Face - Update and LargePersonGroup Person - UpdateFace.

lebronJ replied:
Consider to fix this later

huxuan replied:
resolved

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/PersonOperations.cs:
```diff
@@ -23,20 +23,20 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face
+    /// PersonOperations operations.
     /// </summary>
-    public partial class Person : IServiceOperations<FaceAPI>, IPerson
+    public partial class PersonOperations : IServiceOperations<FaceAPI>, IPersonOperations
```
s/Person/PersonGroupPerson/g

lebronJ replied:
Done

huxuan commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Generated/Face/PersonOperationsExtensions.cs:
```diff
@@ -18,9 +18,9 @@ namespace Microsoft.Azure.CognitiveServices.Vision.Face
+    /// Extension methods for PersonOperations.
     /// </summary>
-    public static partial class PersonExtensions
+    public static partial class PersonOperationsExtensions
```
s/Person/PersonGroupPerson/g

lebronJ replied:
Done

shutchings committed:
Machine Learning Compute: Add optional deleteAll parameter (#3959)

* Add deleteAll query parameter

* Remove unnecessary package tags

ro-joowan committed:
[ADLA] - Adding Scope APIs/Properties, existing Catalog APIs, and existing Account APIs (#3960)

* [CODE GEN] - ADLA ACCOUNT

* [CODE GEN] - ADLA CATALOG

* [CODE GEN] - ADLA JOB

* [VERSION] - ADLA 3.1.2-preview to 3.2.3-preview

* [CHANGELOG] - ADLA Update

* Patching OData type for Job LIST

* Updating tests and adding SessionRecords

* Minor test revision

* Updating Catalog tests and SessionRecords

* Updating Catalog test and SessionRecord

* Adding metadata files from Code Generation

* Adding metadata files from Project Build

* [Second] Adding metadata files from Code Generation

Forgot about resource-manager code gen

* [Second] Adding metadata files from Project Build

It looks like the props file wasn't fully updated, so I ran msbuild
build.proj again before building the project via the IDE

ro-joowan committed:
[ADLS] - Adding existing Account APIs (#3961)

* [CODE GEN] - ADLS ACCOUNT

* [VERSION] - ADLS 2.3.0-preview to 2.3.3-preview

* [CHANGELOG] - ADLS Update

* Skipping flaky test

* Updating tests and adding SessionRecords

* Cleaning up comments in ADLS data-plane cs files

* Adding metadata files from Code Generation

* Adding metadata files from Project Build

kc786 committed:
Added two new operations for Management Consumption (#3972)

* Added two new operations for Management Consumption
1. Reservation Summaries
2. Reservation Details

* updated  *.csproj and AssemblyInfo.cs files with the new version of the SDK.

* Perfrom steps mentioned by shahabhijeet in comments
Added package release notes

* AzureSdk.RP.props file changes

yuanliu-ms committed:
SDK update for ADF ExecuteSsisPackageActivity (#3936)

* SDK update for ADF ExecuteSsisPackageActivity

* Update 1

* Update SDK for ExecuteSSISPackageActivity [Iteration2]

* Revert "Update SDK for ExecuteSSISPackageActivity [Iteration2]"

This reverts commit 4d8be60805ce082d565a22f2d20b8743e2629b84.

* Update SDK for ExecuteSsisPackageActivity [Iteration2]

* SDK update for ADF ExecuteSsisPackageActivity [Iteration3]

* SDK update for ExecuteSsisPackageActivity in ADF

* Regenerate SDK

* Use generate.cmd to generate SDK

* Add AzSdk.RP.props file

JCookit committed:
Changed assembly and package name to official name. (#3984)

hyonholee committed:
Adding Priority property and PUT operation to VmssVM. (#3989)

ro-joowan committed:
[ADLA - ADLS] - Non-functional, minor fixes (comments and changelog) (#3990)

* Test comment fix

* ADLS changelog fix

* Comment fixes

shahabhijeet committed:
Update README.md

Updating readme for building the repo.

irrogozh committed:
Generated SDK for Connection monitor APIs + tests (#3988)

* Generated SDK for Connection monitor APIs + tests

* Added metadata

* Regenerated SDK with the new version of Autorest

* Changes based on CR

pcostantini committed:
[CognitiveServices] LUIS Programmatic SDK (#3987)

* LUIS Programmatic SDK - Updates based on latest Spec

* Add documentation for custom classes

* Updates based on latest Spec changes and Tests

* Cognitive Services Language SDKs directory re-structure

* Model CS  Language projects similar to other CS projects

bitsforthought committed:
[Azure Search] Upgrade Management SDK to version 2.0 (#3993)

* [Azure Search] Adding back missing tools and fixing build warnings

* [Azure Search] Regenerate mgmt SDK with latest version of AutoRest

* [Azure Search] Updating session records for management SDK tests

* [Azure Search] Adding the management SDK to the list of support Nuget packages

* [Azure Search] Adding useful script for regenerating management SDK

The stock generate.cmd script takes way too many parameters to be useful
when iterating on a private Swagger spec.

* [Azure Search] Generating new code for PATCH services and adding tests

Also re-recording all session records.

* [Azure Search] Adding support for Operations API

Added new generated code, plus a scenario test and corresponding session
record.

* [Azure Search] Bumping major version number of management SDK

A major version bump is necessary because the behavior of
Services.CreateOrUpdate has changed. Specifically, service creation and
scaling is now synchronous instead of asynchronous. Also, there is new API
surface area that represents breaking changes from the .NET point of view
(i.e. -- new methods added to existing interfaces).

* Fixing generate.cmd to point to the correct azure-rest-api-specs branch

Since the refactor, the default branch is master, not current.

* [Azure Search] Regenerating management SDK from the latest spec

This includes some minor doc changes.

bitsforthought committed:
[Azure Search] Fixing management SDK package name (#3996)

shahabhijeet committed:
Enabling multiple headers for LRO (#4002)

* Refator LRO to support multiple headers in a requests
* Refactoring check errors. Fixes applied for failing tests for autorest
* Adding comments and code cleanup
* more documentation for LRO rules applicable for each operation
* Updating release notes

chandrasekarsrinivasan committed:
Updating to 2017-11-01 for Network SDK  (#3999)

* Updating version number for Network SDK

* generate Network SDK for 11-01

* Update release notes

kalyanaj committed:
Event Grid: Regenerated the SDK and re-recorded the tests based on the newer API version (#4000)

* Regenerated the SDK and re-recorded the tests based on the newer API version.

* Re-recorded the tests.

* Updated package release notes to reflect this set of changes.

* Re-recorded the tests.

vinayada1 committed:
Pstestfail (#4003)

* Update release notes

* Generated code

hbzhxying committed:
DataFactory: convert all securestring to secretbase in Linked Service (#3997)

* change all securestring to secret base in linked service in sdk

* update the changelog.md

* use generate.cmd

* update metedata

shahabhijeet committed:
Update Auth library to relax ADAL references as range (#4006)

hvermis committed:
Updating ADF SDK version for the release (#4007)

* Updating ADF SDK version for the release

* Updating ADF SDK version for the release

avneeshraiusit committed:
Fixing wrong discriminator value for AzureFabricCreationInput (#3998)

* Fixing wrong discriminator value for AzureFabricCreationInput
* add metadata files

lebronJ committed:
Regenerate Face/ComputerVision SDK with latest spec and update the unit tests, separate Face/CV aligning with other Vision APIs.

lebronJ commented:
Closed since this has already been merged to Azure/azure-sdk-for-net.
